README.md

```diff
@@ -410,37 +316,26 @@ enum Mixed {
 export enum Zoo { lion, tiger, bear };
 export default enum { up, down, left, right };
 
-// All enum declaration constructors inherit from a built in %Enum% object with common 
-// static methods:
-
-// You can test for membership using `%Enum%.has()`:
-Numbers.has(0) === true; // for underlying values
-Numbers.has("one") === true; // for enum member names
-Numbers.has(Numbers.two) === true; // for enum values
-Numbers.has(Numbers(9)) === false; // even though coerced
-Numbers.has(9) === false;
-Numbers.has(Color.blue) == false; // even though the underlying values are the same
-
-// You can test for enum branding using `%Enum%.is()`:
-Numbers.is(0) === false;
-Numbers.is("one") === false;
-Numbers.is(Numbers.two) === true;
-Numbers.is(Numbers(9)) === true; // even though not defined
-Numbers.is(Color.blue) === false;
-
-// You can lookup an enum value by key or value using `%Enum%.forKey()` and 
-// `%Enum%.forValue()`, respectively:
-enum AlphaBeta {
+// You can test for name membership using `Enum.has()`
+Enum.has(Numbers, "one") === true
```

Comment:
The enum object's \[\[Prototype]] is `null`.

Reply:
Even with the extends keyword? That suggests “extends” isn’t the best keyword to use :-/

Reply:
Perhaps, but `enum` is not `class`. @rwaldron's proposal used `via` but if I had to choose a keyword other than `extends` it would probably be either `of` or `as`:

```js
enum Colors of Number { Red, Green, Blue }
enum Colors as Number { Red, Green, Blue }
```

Alternatively we could use C# style:

```js
enum Colors: Number { Red, Green, Blue }
```

However I think I'll stick with `extends` for now and hold off on further bikeshedding until later.

Reply:
`via` definition: "by way of; through." or "by means of."

```
enum Flags via bits {A, B, C}
```

Means: "Define an enum named Flags, whose values are defined _via_ the function 'bits'".

In your examples above, `Colors` doesn't not _extend_ `Number`: it derives its values by passing its keys "through" `Number[Symbol.toEnum]`, or via the `Number[Symbol.toEnum]` function.

Reply:
My reticence towards `via` is that its not a commonly used keyword in major programming languages. If `extends` seems problematic, what do you think of a compromise using `of`, since it is already a keyword in ES, and closely approximates both of our intended meanings?

Reply:
There’s also `from` and `with`

Reply:
> There's also `from` and `with`

Linguistically, `from` and `with` have different implications than `extends`, `via`, or `of` that 
make them a bit less informative:

```js
// `extends` implies enum members are `X` by default.
enum DaysOfTheWeek extends X { SUNDAY, ... }
// - Works well if `X` is a built-in constructor function: 
//    enum E extends Number { ... }
// - Somewhat dissonant if `X` is a mapping function:
//    enum E extends bits { ... }

// `via` implies enum members are piped through `X`.
enum DaysOfTheWeek via X { SUNDAY, ... }
// - Works well if `X` is a mapping function:
//    enum E via bits { ... }
// - Somewhat dissonant if `X` is a built-in constructor function:
//    enum E via Symbol { ... }

// `of` implies enum members are by default `X` constituents,
// or that `X` provides the set of results.
enum DaysOfTheWeek of X { SUNDAY, ... }
// - Works well if `X` is a built-in constructor function:
//    enum E of Symbol { ... }
// - Works well if `X` is a mapping function:
//    enum E of bits { ... }

// `from` implies members are read from an `X` container, 
// but possibly could imply that `X` is some kind of provider of values.
enum DaysOfTheWeek from X { SUNDAY, ... }
// - Works well if `X` is a mapping function:
//    enum E from bits { ... }
// - Somewhat dissonant if `X` is a built-in constructor function:
//    enum E from Symbol { ... }

// `with` implies `X` is some kind of augmentation (feels like a mixin)
enum DaysOfTheWeek with X { SUNDAY, ... }
// - Very dissonant as either a mapping function or a built-in constructor:
//    enum E with Symbol { ... }
//    enum E with bits { ... }
```

As these are my own interpretations, I'd be interested in everyone else's take. That said, `of` is starting to grow on me.

Reply:
to me, both "of" and "from" have the same connotation - the one you see from "of".

Reply:
`from` indicates an origin ("I'm leaving from work"), source of provenance ("I'm from the suburbs"), or a starting point in a range ("From one to ten"). In ECMAScript, `from` is used to indicate a module source (`import { x } from "y"`) or to construct an array from a sequence of elements (`Array.from(x)`).

So `enum E from String {}` reads to me either as:
- As an origin or source of provenance: "The enum E is from String"
- As a starting point: "Create the enum E starting at String". 

Neither interpretation matches the behavior, which dissuades me from `from`.

`of` indicates a constituency ("one of many") or relationship ("friend of a friend"). In ECMAScript, `of` is used to pick individual elements from a sequence (`for..of`).

So `enum E of String {}` reads to me as: "Each enum member of E is of the type String", while `enum E of bits {}` reads to me as: "Each enum member of E is a constituent of 'bits'".

Both interpretations match the behavior.

Reply:
I like `of` because it's short, but I'm just going to point out that it's essentially a heritage clause from the grammar perspective, I'd imagine. Is it really worth having different keywords for the slightly different semantics the clause has for each declaration type?

Reply:
Or another way: if I've already used classes, why would my first instinct not to try and use `extends`, not understanding the underlying mechanism?

Reply:
`from` also indicates an origin - "i built that from nothing", "from rags to riches", "that tree grew from a sapling".

"of", to me, also indicates that the items inside the curly braces are of that type - `enum E of String { … }` tells me there must be strings inside there, and `enum E of Number { … }` tells me there has to be numbers in there (but in all cases, it's string names, not actual values)

Reply:
> `from` also indicates an origin - "i built that from nothing", "from rags to riches", "that tree grew from a sapling".

That's precisely what I was describing here:

> `from` indicates an origin ("I'm leaving from work"), source of provenance ("I'm from the suburbs"), or a **starting point in a range** ("From one to ten").

As to your other point:

> of", to me, also indicates that the items inside the curly braces are of that type - `enum E of String { … }` tells me there must be strings inside there, and `enum E of Number { … }` tells me there has to be numbers in there (but in all cases, it's string names, not actual values)

The subject for `of` in those cases are the _values_ of the enumeration members, not the member names. Barring the user overriding that behavior with an explicit initializer, this is the exact semantics:

```js
enum A of Number { zero, one, two }
typeof A.zero === "number"
typeof A.one === "number"
...
enum B of String { zero, one, two}
typeof B.zero === "string"
typeof B.one === "string"
...

enum C of Symbol { zero, one, two }
typeof C.zero === "symbol"
typeof C.one === "symbol"
...
```

Reply:
> My reticence towards via is that its not a commonly used keyword in major programming languages

Lol... So what? The idea of creating enum values the way I describe them in my proposal isn't common in programming languages either, but your changes to my idea aren't semantically different enough to warrant giving it a new syntax keyword that doesn't make sense. 


> what do you think of a compromise using of, since it is already a keyword in ES, and closely approximates both of our intended meanings?

No. There is no requirement that we use an ES keyword.

```js
enum We can do whatever we want here {
```


@ljharb 

> There’s also from and with

- "from" indicates a starting point, whether in time, space or materials. This isn't that. 
- "with" indicates possession or accompaniment... also it has too much baggage in JS


@rbuckton 

> ```js
> // `via` implies enum members are piped through `X`.
> enum DaysOfTheWeek via X { SUNDAY, ... }
> // - Works well if `X` is a mapping function:
> //    enum E via bits { ... }
> // - Somewhat dissonant if `X` is a built-in constructor function:
> //    enum E via Symbol { ... }
```

> Works well if `X` is a mapping function:
> Somewhat dissonant if `X` is a built-in constructor function:

The semantics of the thing on the right of `via` and to the left of `{` is _mapping_, whether its by calling an @@toEnum method, or calling directly.